megamek/src/megamek/common/autoresolve/acar/phase/EndPhase.java

```diff
@@ -66,10 +68,29 @@ private void checkWithdrawingForces() {
             // If the game is over, no need to withdraw
             return;
         }
+
         var forcedWithdrawingUnits = getSimulationManager().getGame().getActiveDeployedFormations().stream()
             .filter(f -> f.moraleStatus() == Formation.MoraleStatus.ROUTED || f.isCrippled())
             .filter(f -> !(f.isType(SBFElementType.BA) || f.isType(SBFElementType.CI)))
-            .toList();
+            .collect(Collectors.toSet());
+
+        for (var order : getContext().getOrders()) {
+            if (order.isEligible(getContext())) {
+                if (order.getOrderType().equals(OrderType.WITHDRAW_IF_CONDITION_IS_MET)) {
+                    var formations = getContext().getActiveFormations(order.getOwnerId());
+                    forcedWithdrawingUnits.addAll(formations);
+                    order.setConcluded(true);
+                } else if (order.getOrderType().equals(OrderType.FLEE_NORTH)) {
+                    var formations = getContext().getActiveFormations(order.getOwnerId());
+                    forcedWithdrawingUnits.addAll(formations);
+                    order.setConcluded(true);
+                } else if (order.getOrderType().equals(OrderType.FLEE_SOUTH)) {
+                    var formations = getContext().getActiveFormations(order.getOwnerId());
+                    forcedWithdrawingUnits.addAll(formations);
+                    order.setConcluded(true);
```

Comment:
It looks like this could be a single block supporting every `OrderType.FLEE_*` type.